server/mentions.go

```diff
@@ -0,0 +1,311 @@
+func NewNotificationsParser(api plugin.API, msteamsAppClient msteams.Client) *NotificationsParser {
+	return &NotificationsParser{
+		PAPI:             api,
+		msteamsAppClient: msteamsAppClient,
+	}
+}
+
+func (p *NotificationsParser) ProcessPost(post *model.Post) error {
+	mentions := p.extractMentionsFromPost(post)
+	channel, err := p.PAPI.GetChannel(post.ChannelId)
+	if err != nil {
+		return err
+	}
+
+	for _, mention := range mentions {
+		m := &UserNotification{
+			Trigger: mention,
+			Post:    post,
+			Channel: channel,
+		}
+
+		switch mention {
+		case "@here":
+			fallthrough
+		case "@channel":
+			fallthrough
+		case "@all":
+
+		default:
+			user := p.isUserMention(strings.TrimPrefix(mention, "@"))
+			if user != nil {
+				m.User = user
+			} else {
+				group := p.isGroupMention(strings.TrimPrefix(mention, "@"))
+				if group != nil {
+					m.Group = group
+				}
+			}
+
+			if m.User == nil && m.Group == nil {
+				p.PAPI.LogError("Failed to find user or group for metnion", "mention", mention)
+				continue
+			}
+		}
+
+		p.Notifications = append(p.Notifications, m)
+	}
+
+	if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
+		p.Notifications = append(p.Notifications, &UserNotification{
+			Trigger: post.Message,
+			Post:    post,
+			Channel: channel,
+		})
+	}
+
+	for _, notification := range p.Notifications {
+		p.PAPI.LogDebug("Processed mention", "notification", *notification)
+	}
+
+	return nil
+}
+
+func (p *NotificationsParser) extractMentionsFromPost(post *model.Post) []string {
+	// Regular expression to find mentions of the form @username
+	mentionRegex := regexp.MustCompile(`@[a-zA-Z0-9._-]+`)
+	return mentionRegex.FindAllString(post.Message, -1)
+}
+
+func (p *NotificationsParser) isUserMention(mention string) *model.User {
+	user, err := p.PAPI.GetUserByUsername(mention)
+	if err != nil {
+		return nil
+	}
+	return user
+}
+
+func (p *NotificationsParser) isGroupMention(mention string) *model.Group {
+	group, err := p.PAPI.GetGroupByName(mention)
+	if err != nil {
+		return nil
+	}
+	return group
+}
+
+func (p *NotificationsParser) SendNotifications() error {
+	for _, userNotification := range p.Notifications {
+		if err := p.SendNotification(userNotification); err != nil {
+			p.PAPI.LogError("Failed to send notification", "error", err.Error())
+		}
+	}
+	return nil
+}
+
+func (p *NotificationsParser) SendNotification(notification *UserNotification) error {
+	// Send notifications for direct and group messages
+	if notification.Channel.Type == model.ChannelTypeDirect || notification.Channel.Type == model.ChannelTypeGroup {
+		return p.sendChannelNotification(notification, false)
+	}
+
+	if notification.User != nil {
+		return p.sendUserNotification(notification)
+	}
+
+	if notification.Group != nil {
+		return p.sendGroupNotification(notification)
+	}
+
+	switch notification.Trigger {
+	case "@here":
+		return p.sendChannelNotification(notification, true)
+	case "@channel":
```

Comment:
You can have multiple expressions in a single case.  `fallthrough` is not needed here.

Reply:
@fmartingr it needs to be `case "@channel", "@all":`